File: leetcode/javascript/1-two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
            var temp = {};
            var result = [];
            nums.forEach(function(item,index,nums){
                if(temp[item]!==undefined){
                    result= [temp[item],index];
                }
                temp[target-item] = index;
            });
            return result;
        };

File: leetcode/java/1-two_sum.java

/**
*
* 使用HashMap
*/
public class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> temp = new HashMap<>();
        int result[] = new int[2];
        for (int i = 0;i<nums.length; i++){
            if (temp.containsKey(target- nums[i])){
                result[0] = temp.get(target- nums[i]);
                result[1] = i;
            }
            temp.put(nums[i],i);
        }
        return result;
    }
}

/**
*
* 原始方法
*/
public class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        for(int i = 0;i<nums.length-1;i++){
            for (int j=i+1 ;j<nums.length;j++){
                if (nums[i]+nums[j]==target){
				    result[0] = i;
				    result[1] = j;
			    }
            }
        }
        return result;
    }
}